main.tf

```diff
@@ -146,6 +146,15 @@ resource "aws_s3_bucket" "origin" {
   }
 }
 
+resource "aws_s3_bucket_public_access_block" "origin" {
+  count                   = ! local.using_existing_origin && var.block_origin_public_access_enabled ? 1 : 0
+  bucket                  = module.origin_label.id
```

Comment:
@juliusdanielherreraglomm apologies I didn't catch this before, but not using the `aws_s3_bucket.origin.id` (the bucket name) here might cause this resource to try and be created before the `aws_s3_bucket.origin` resource which would result in an error. Can you update this to use the reference to the bucket name instead of the `origin_label.id`?

Reply:
@Gowiem Good catch. This similar to the `aws_s3_bucket_policy.default` resource which also uses the bucket name and needs to ensure that the bucket is created before. In this case it is ensured by using `local.bucket` which has a dependency to the bucket.

For consistency reasons I suggest to use the same way. I modified the PR. Could you take a look?

Reply:
Looks good @juliusdanielherreraglomm -- Thanks for the update.